modules/process-testing/pages/task.adoc

```diff
@@ -1,3 +1,106 @@
+        Task userTask = processInstance.getFirstPendingUserTask("aTask"); <1>
+        Contract contract = createContract(); <2>
+
+        // Check the candidates of a task
+        Set<User> candidates = userTask.getCandidates(); <3>
+        assertThat(candidates).extracting(User::getUserName).containsExactly(walter.getUserName());
+
+        // Assign a task to a user
+        userTask.assignTo(walter); <4>
+
+        // Execute an assigned task, with or without a contract.
+        userTask.execute(); <5>
+        userTask.execute(contract);
+
+        // Assign and execute a task, with or without a contract.
+        userTask.execute(walter); <6>
+        userTask.execute(walter, contract);
+
+        // Use the provided predicates to ensure that a taks has been correctly executed before to continue.
+        await().until(userTask, hasBeenExecutedBy(walter) <7>
+                .and(taskArchived())
+                .and(taskCompleted()));
+    }
+}
+----
+<1> Retrieve the pending task _aTask_ from the process instance.
+<2> Create a contract from a dedicated method to execute the task (more details about contract creation xref:contract.adoc[here]).
+<3> Retrieve the candidates of the task (i.e the users that can be assigned to the task), and ensure that walter is the only candidate.
+<4> Assign the task to a user.
+<5> Execute the task with or without a contract. The task must have been already assigned.
+<6> Assign and execute the task with or without a contract. If the task is already assigned to an other user, it will be reassigned to the new user before to be executed.
+<7> Wait until the task is completed before to continue. Mor details about task predicates in the dedicated section xref:_task_predicates[bellow].
+
+== Task predicates
+
+In order to make asynchronous assertions on processes (using for example http://www.awaitility.org/[Awaitility]), some convenient predicates come with the Bonita test tookit. It allows to ensure in a scenario that the system is in the expected state before to continue. +
+For example, it is possible to use the predicate `taskCompleted()` on a user task to ensure that a task has been correctly executed before to continue. Awaitility (or any other asynchronous library) will check this predicate at a given frequency, and will throw an error if the timeout is reached (i.e the case has not been started in time).
+
+[source, Java]
+----
+@BonitaTests
+class ProcessIT {
+
+    @Test
+    void atask_predicates(BonitaTestToolkit toolkit) {
+        User walter = toolkit.getUser("walter.bates");
+        User helen = toolkit.getUser("helen.kelly");
+        ProcessDefinition processDef = toolkit.getProcessDefinition("myProcess");
+        ProcessInstance processInstance = processDef.startProcessFor(walter);
+        Task userTask = processInstance.getFirstPendingUserTask("aTask");
+
+
+        await().until(userTask, isNotAssigned()); <1>
```

Comment:
```suggestion
        await("Verify that the user task has no assignee").until(userTask, isNotAssigned()); 
```
Use the awaitility DSL instead of the asciidoc callout ?

Reply:
I don't think so, if the description is out of the code bloc then it becomes searchable (we do not index code blcoks).